code/modules/admin/topic.dm

```diff
@@ -512,6 +512,7 @@
 			return
 
 		M.forceMove(pick(GLOB.prisonwarp))
+		M.AddComponent(/datum/component/imprisoned)
```

Comment:
how would you undo this after?

Reply:
> how would you undo this after?

See the imprisoned tracker, it does the same thing as that itembound component I did earlier this year. Whenever the move_tracker updates, it callbacks to a proc that verifies if the area the prisoner is currently in is still that same area that we have defined in code (admin prison cells). If the areas mismatch, we delete the component and remove the trait.